File: ta-java/ShoutyReportProcessor.java
import org.w3c.dom.Document;

import java.util.List;

class ShoutyReportProcessor {
    List<MileageClaim> mileageClaims;
    ShoutyStatsService statsService = new ShoutyStatsService();

    public ShoutyReportProcessor(List<MileageClaim> mileageClaims) {
        this.mileageClaims = mileageClaims;
    }

    public List<EcoStat> process() {
        List<EcoStat> results = new ArrayList<EcoStat>();

        for (MileageClaim claim : mileageClaims) {
            String requestXml = "<Customer id=\"" + claim.customerID + "\"/>";
            String responseXml = statsService.getRevenueForCustomer(requestXml);
            Document responseDocument = XmlHelper.parse(responseXml);

            double revenue = Double.parseDouble(responseDocument.getDocumentElement().getAttribute("revenue"));
            float revenuePerMile = (float) revenue / (float) claim.miles;

            results.add(new EcoStat(claim.name, revenuePerMile));
        }

        return results;
    }
}

File: ta-csharp/ShoutyReportProcessor.cs
using System.Collections.Generic;
using System.Xml;

class ShoutyReportProcessor
{
    IList<MileageClaim> mileageClaims;
    ShoutyStatsService statsService = new ShoutyStatsService();

    public ShoutyReportProcessor(IList<MileageClaim> mileageClaims)
    {
        this.mileageClaims = mileageClaims;
    }

    public List<EcoStat> Process()
    {
        List<EcoStat> results = new List<EcoStat>();

        foreach (var claim in mileageClaims)
        {
            string requestXml = "<Customer id=\"" + claim.CustomerID + "\"/>";
            string responseXml = statsService.GetRevenueForCustomer(requestXml);

            var responseDocument = new XmlDocument();
            responseDocument.LoadXml(responseXml);

            var revenue = decimal.Parse(responseDocument.DocumentElement.Attributes["revenue"].Value);
            var revenuePerMile = (float)revenue / (float)claim.Miles;

            results.Add(new EcoStat(claim.Name, revenuePerMile));
        }

        return results;
    }
}
